todo_app/app.py

```diff
@@ -1,17 +1,40 @@
 from flask import Flask, render_template, request, redirect
 from operator import itemgetter
-import pytest, datetime
+import pytest, datetime, os, requests, json
+from flask_login import LoginManager, login_required, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+from todo_app.user import User
 
 from todo_app.flask_config import Config
 from todo_app.data.todo_items import ToDoCard, ViewModel, get_todo_cards, move_todo_card, create_todo_card, delete_todo_card
 
+
+login_manager = LoginManager()
+
+@login_manager.unauthorized_handler
+def unauthenticated():
+    github_client =  WebApplicationClient(os.environ.get('CLIENTID'))
+    github_redirect = github_client.prepare_request_uri("https://github.com/login/oauth/authorize")
+
+    return redirect(github_redirect) 
+
+@login_manager.user_loader
+def load_user(github_user):
+    
+    return User(github_user)
+
+
 def create_app():  
     app = Flask(__name__)
+    app.secret_key = os.urandom(24)
```

Comment:
This will work here but would be better as an environment variable, otherwise:
* If we tried to scale the server by running multiple instances, they'll all have different secrets and users will lose their session if they end up talking to different servers
* Every time the server is restarted all sessions will be invalidated